windows/src/engine/keyman32/kmhook_getmessage.cpp

```diff
@@ -412,6 +420,32 @@ void ProcessWMKeymanControlInternal(HWND hwnd, WPARAM wParam, LPARAM lParam)
 	}
 }
 
+void
+ProcessWMKeymanControl(WPARAM wParam, LPARAM lParam) {
+
+  switch (wParam) {
+  case KMC_PROFILECHANGED:
+    if (!Globals::IsControllerThread(GetCurrentThreadId())) {
+      break;
+    }
+    WORD wAtom = HIWORD(lParam);
+    char atomStr[128];
+    if (GetAtomName(wAtom, atomStr, 128)) {
+      // Is this a keyman keyboard check c_clsidKMTipTextService
+      WCHAR clsidstr[40];
+      StringFromGUID2(c_clsidKMTipTextService, clsidstr, _countof(clsidstr));
+      PWSTR atomWstr = strtowstr(atomStr);
```

Comment:
Given `c_clsidKMTipTextService` is a constant, can we declare this as a constant WCHAR and avoid doing this conversion every keystroke? (That then means the leak I noted moments ago will be a non-issue too)

Reply:
My bad, my note was a little unclear and slightly wrong.

1. We can declare an additional `const char cs_clsidKMTipTextService[] = "{XXXX-XXXX...}";` in the kmtip_guids header/cpp. (check the format the StringFromGUID2 returns and match that, including casing!)
2. Then we can just use the equiv of l.438: `if (strstr(atomstr, cs_clsidKMTipTextService)) {` and avoid all the conversion and allocation code :grin:

Reply:
For 1. We could assign the  cs_clisdKMtipTextService[] in the `kmtip_guids.cpp` I think using the `StringFromGUID2` to ensure it matches.... but that is probably overkill.  Seeing as you are never going to update the guid are you only add new ones

Reply:
Yep, overkill. One time code review check when we update kmtip_guids.cpp!